feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/presentation/referenda/vote/setup/common/SetupVoteViewModel.kt

```diff
@@ -36,40 +31,41 @@ import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.WithFeeL
 import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.awaitDecimalFee
 import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.connectWith
 import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.create
-import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.mapFeeToParcel
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.launch
 
-class SetupVoteReferendumViewModel(
+abstract class SetupVoteViewModel(
     private val feeLoaderMixinFactory: FeeLoaderMixin.Factory,
     private val assetUseCase: AssetUseCase,
     private val amountChooserMixinFactory: AmountChooserMixin.Factory,
     private val interactor: VoteReferendumInteractor,
-    private val payload: SetupVoteReferendumPayload,
+    private val payload: SetupVotePayload,
     private val resourceManager: ResourceManager,
     private val router: GovernanceRouter,
     private val validationSystem: VoteReferendumValidationSystem,
     private val validationExecutor: ValidationExecutor,
-    private val referendumFormatter: ReferendumFormatter,
     private val locksChangeFormatter: LocksChangeFormatter,
     private val convictionValuesProvider: ConvictionValuesProvider,
     private val locksFormatter: LocksFormatter,
 ) : BaseViewModel(),
     WithFeeLoaderMixin,
     Validatable by validationExecutor {
 
-    private val selectedAsset = assetUseCase.currentAssetFlow()
+    abstract val title: Flow<String>
+
+    protected val selectedAsset = assetUseCase.currentAssetFlow()
         .shareInBackground()
 
     private val voteAssistantFlow = interactor.voteAssistantFlow(payload.referendumId, viewModelScope)
 
     override val originFeeMixin = feeLoaderMixinFactory.create(selectedAsset)
 
-    private val validatingVoteType = MutableStateFlow<VoteType?>(null)
+    protected val validatingVoteType = MutableStateFlow<VoteType?>(null)
```

Comment:
Looks like this is something related to `VoteReferendum` subclass and not the base class

Reply:
Yes but it's easier to keep it here to avoid more complicated refactoring